samples/OriginalUriBaseIds.sarif

```diff
@@ -0,0 +1,77 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "properties": {
+    "comment": "This sample demonstrates the use of originalUriBaseIds."
+  },
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "SarifSamples"
+        }
+      },
+      "originalUriBaseIds": {
+        "PROJECTROOT": {
+          "description": {
+            "text": "The root directory for all project files."
+          },
+          "properties": {
+            "comment": "The SARIF producer has chosen not to specify a URI for PROJECTROOT. See §3.14.14, NOTE 1, for an explanation."
+          }
+        },
+        "SRCROOT": {
+          "uri": "src",
+          "uriBaseId": "PROJECTROOT",
+          "description": {
+            "text": "The root of the source tree."
+          },
+          "properties": {
+            "comment": "SRCROOT is expressed relative to PROJECTROOT."
+          }
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "A result outside the source tree."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "README.md",
+                  "uriBaseId": "PROJECTROOT",
+                  "properties": {
+                    "comment": "If PROJECTROOT is C:\\project, this file location resolves to C:\\project\\README.md"
+                  }
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0002",
+          "message": {
+            "text": "A result in a source file."

```

Comment:
>source file [](start = 35, length = 11)

hm, shall we power up this sample and add a reference to a directory in addition to distinct files? is this a concern that belongs in a different sample?

Reply:
Good idea, done.

---
In reply to: [414794276](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414794276) [](ancestors = 414794276)